src/engine/parser/lexer/error.rs

```diff
@@ -0,0 +1,188 @@
+            SyntaxErrorCode::UnexpectedEof{..} => "Unexpected end of template.",
+            SyntaxErrorCode::UnclosedBracket{..} => "Unclosed bracket.",
+            SyntaxErrorCode::UnclosedComment{..} => "Unclosed comment.",
+            SyntaxErrorCode::UnclosedBlock{..} => "Unclosed block.",
+            SyntaxErrorCode::UnclosedVariable{..} => "Unclosed variable.",
+        }
+    }
+}
+
+impl Display for SyntaxErrorCode {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        try!(write!(f, "{}", self.description()));
+
+        match *self {
+            SyntaxErrorCode::Unreachable {
+                reason: ref r,
+                cursor: ref c
+            } => {
+                write!(f, " {reason} at {cursor}.",
+                    reason = r,
+                    cursor = c)
+            },
+            SyntaxErrorCode::UnexpectedCharacter {
+                character, ref cursor
+            } => {
+                write!(f, " found '{c}' at {cursor}.",
+                    c = character, cursor = cursor)
+            },
+            SyntaxErrorCode::UnexpectedBracket {
+                ref cursor, ref bracket
+            } => {
+                write!(f, " Unexpected {bracket:?} at {cursor}.",
+                    cursor = cursor, bracket = bracket)
+            },
+            SyntaxErrorCode::UnexpectedEof {
+                reason: ref r,
+                cursor: ref c
+            } => {
+                write!(f, " {reason} at {cursor}.",
+                    reason = r,
+                    cursor = c)
+            },
+            SyntaxErrorCode::UnclosedBracket {
+                ref cursor, ref bracket, ref bracket_before, line_before
+            } => {
+                write!(f, " Unclosed {b_before:?} from line\
+                                {line_before} but found {b:?} at {cursor}.",
+                    cursor = cursor,
+                    b = bracket,
+                    b_before = bracket_before,
+                    line_before = line_before)
+            },
+            SyntaxErrorCode::UnclosedComment {
+                ref cursor
+            } => {
+                write!(f, " At {cursor}.",
+                    cursor = cursor)
+            },
+            SyntaxErrorCode::UnclosedBlock {
+                ref cursor
+            } => {
+                write!(f, " At {cursor}.",
+                    cursor = cursor)
+            },
+            SyntaxErrorCode::UnclosedVariable {
+                ref cursor
+            } => {
+                write!(f, " At {cursor}.",
+                    cursor = cursor)
+            },
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub enum LexerErrorCode {
+    Unreachable {
+        reason: String
+    },
+    MissingExtensions,
+    PatternRegexError,
+    _InvalidPatternMatch,
+    InvalidValue {
+        value: String
+    },
+    _InvalidState,
+    SyntaxError,
```

Comment:
I'll remove it for now. I guess not many error codes will be left after this cleanup... ;-)